CMakeLists.txt

```diff
@@ -5,18 +5,44 @@ cmake_minimum_required(VERSION 3.11)
 set(CCF_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 include(${CCF_DIR}/cmake/preproject.cmake)
 
+# Deduce project version from git environment
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
+  FIND_PACKAGE(Git)
+  execute_process(
+    COMMAND git describe --tags
+    OUTPUT_VARIABLE "CCF_VERSION"
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  execute_process(
+    COMMAND bash "-c" "git describe --tags --abbrev=0 | cut -d'-' -f1 | tr -d v"
+    OUTPUT_VARIABLE "CCF_RELEASE_VERSION" OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  # Check that git release tag follows semver
+  execute_process(
+    COMMAND
+      bash "-c"
+      "[[ ${CCF_RELEASE_VERSION} =~ ^([[:digit:]])+(\.([[:digit:]])+)*(-.*)?$ ]]"
+    RESULT_VARIABLE "TAG_IS_SEMVER"
+  )
+  if(NOT ${TAG_IS_SEMVER} STREQUAL "0")
+    message(
+      FATAL_ERROR "git tag \"${CCF_RELEASE_VERSION}\" does not follow semver"
+    )
+  endif()
+else()
+  set(CCF_VERSION "unknown")
+  set(CCF_RELEASE_VERSION "0.0")
+endif()
+
 project(
   ccf
-  VERSION 0.11
+  VERSION ${CCF_RELEASE_VERSION}
   LANGUAGES C CXX
 )
 
 set(ENV{BETTER_EXCEPTIONS} 1)
 
-set(CCF_PROJECT_NAME_SUFFIX "")
-set(CCF_PROJECT_NAME "${PROJECT_VERSION}${CCF_PROJECT_NAME_SUFFIX}")
```

Comment:
What was `CCF_PROJECT_NAME_SUFFIX` used for?

Reply:
This was where we could the pre-release identifier (`-rc1` etc) in CMake previously. You can't put it in `project( VERSION ...)`, so we add a manual suffix and dump the whole name for the script to parse + compare to the git tag.

Reply:
@eddyashton That's what I thought too but I couldn't see being used. I believe we don't need it anymore then?

Reply:
This is where it was used, but we don't need it anymore.